File: Front/RHStoreWS/RHStoreComprasBO/OrdenCompraBO.cs
using RHStoreBaseBO.ServiciosWeb;
using RHStoreBaseBO;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RHStoreComprasBO
{
	public class OrdenCompraBO : BaseBO
	{
		public int insertar(DateTime fechaRegistro, DateTime fechaProcesado, DateTime fechaEntregado, DateTime fechaAnulado,
			estado _estado, string dni, string correo, double subtotal, cliente _cliente)
		{
			return OrdencompraWS.insertarOrdenCompra(fechaRegistro, fechaProcesado, fechaEntregado, fechaAnulado, _estado, dni, correo, subtotal, _cliente);
		}

		public int modificar(int IdOrdenCompra, DateTime fechaRegistro, DateTime fechaProcesado, DateTime fechaEntregado, DateTime fechaAnulado,
			estado _estado, string dni, string correo, double subtotal, cliente _cliente)
		{
			return OrdencompraWS.modificarOrdenCompra(IdOrdenCompra, fechaRegistro, fechaProcesado, fechaEntregado, fechaAnulado, _estado, dni, correo, subtotal, _cliente);
		}

		public int eliminar(int idOrdenCompra)
		{
			return OrdencompraWS.eliminarOrdenCompra(idOrdenCompra);
		}

		public BindingList<ordenCompra> listarOrdenCompraTodos()
		{
			return new BindingList<ordenCompra>(OrdencompraWS.listarOrdenCompraTodos());
		}

		public ordenCompra obtenerPorId(int idOrdenCompra)
		{
			return OrdencompraWS.obtenerOrdenCompraPorId(idOrdenCompra);
		}

		public BindingList<ordenCompra> listarOrdenCompraPorEstado(string estado)
		{
			return new BindingList<ordenCompra>(OrdencompraWS.listarOrdenPorEstado(estado));
		}
	}
}

File: Back/SoftRhComprasBO/src/pe/edu/pucp/softrh/compras/bo/OrdenCompraBO.java
package pe.edu.pucp.softrh.compras.bo;

import java.util.ArrayList;
import java.util.Date;
import pe.edu.pucp.softrh.compras.dao.OrdenCompraDAO;
import pe.edu.pucp.softrh.compras.daoimp.OrdenCompraDAOImp;
import pe.edu.pucp.softrh.compras.model.Estado;
import pe.edu.pucp.softrh.compras.model.OrdenCompra;
import pe.edu.pucp.softrh.usuarios.model.Cliente;

public class OrdenCompraBO {
    private OrdenCompraDAO ordenCompraDAO;

    public OrdenCompraBO(){
        this.ordenCompraDAO = new OrdenCompraDAOImp();
    }

    public Integer insertar(Date fechaRegistro, Date fechaProcesado,
            Date fechaEntregado,Date fechaAnulado, Estado estado,String dni,
            String correo, Double subtotal, Cliente cliente) {
        OrdenCompra orden = new OrdenCompra(fechaRegistro,fechaProcesado,fechaEntregado,
                fechaAnulado, estado, dni, correo, subtotal, cliente);
        orden.setIdOrden(ordenCompraDAO.insertar(orden));
        return orden.getIdOrden();
    }

    public Integer modificar(Integer IdOrdenCompra, Date fechaRegistro, Date fechaProcesado,
            Date fechaEntregado,Date fechaAnulado, Estado estado,String dni,
            String correo, Double subtotal, Cliente cliente){
        OrdenCompra orden = new OrdenCompra(fechaRegistro,fechaProcesado,fechaEntregado,
                fechaAnulado, estado, dni, correo, subtotal, cliente);
        orden.setIdOrden(IdOrdenCompra);
        return ordenCompraDAO.modificar(orden);
    }

    public Integer eliminar(Integer idOrden){
        return this.ordenCompraDAO.eliminar(idOrden);
    }

    public ArrayList<OrdenCompra> listarTodos() {
        return this.ordenCompraDAO.listarTodos();
    }
    
    public OrdenCompra obtenerPorId(Integer idOrden) {
        return this.ordenCompraDAO.obtenerPorId(idOrden);
    }
    
    public ArrayList<OrdenCompra> listarPorEstado(String cadena) {
        return this.ordenCompraDAO.listarPorEstado(cadena);
    }
}
